shard-operator-parent/shard-operator-v2/src/test/java/com/redhat/service/smartevents/shard/operator/v2/ManagedBridgeServiceImplTest.java

```diff
@@ -0,0 +1,175 @@
+package com.redhat.service.smartevents.shard.operator.v2;
+
+import java.time.Duration;
+
+import javax.inject.Inject;
+
+import org.awaitility.Awaitility;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import com.redhat.service.smartevents.infra.v2.api.models.OperationType;
+import com.redhat.service.smartevents.infra.v2.api.models.dto.BridgeDTO;
+import com.redhat.service.smartevents.shard.operator.core.providers.GlobalConfigurationsConstants;
+import com.redhat.service.smartevents.shard.operator.core.resources.knative.KnativeBroker;
+import com.redhat.service.smartevents.shard.operator.core.utils.LabelsBuilder;
+import com.redhat.service.smartevents.shard.operator.v2.providers.NamespaceProvider;
+import com.redhat.service.smartevents.shard.operator.v2.resources.ManagedBridge;
+import com.redhat.service.smartevents.shard.operator.v2.utils.Fixtures;
+import com.redhat.service.smartevents.shard.operator.v2.utils.V2KubernetesResourcePatcher;
+
+import io.fabric8.kubernetes.api.model.ConfigMap;
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.fabric8.kubernetes.api.model.Secret;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.quarkus.test.junit.QuarkusTest;
+import io.quarkus.test.kubernetes.client.WithOpenShiftTestServer;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@QuarkusTest
+@WithOpenShiftTestServer
+public class ManagedBridgeServiceImplTest {
+
+    @Inject
+    ManagedBridgeService managedBridgeService;
+
+    @Inject
+    KubernetesClient kubernetesClient;
+
+    @Inject
+    NamespaceProvider namespaceProvider;
+
+    @Inject
+    V2KubernetesResourcePatcher kubernetesResourcePatcher;
+
+    @BeforeEach
+    public void beforeEach() {
+        kubernetesResourcePatcher.cleanUp();
+    }
+
+    @Test
+    public void createManagedBridge() {
+        BridgeDTO bridgeDTO = Fixtures.createBridge(OperationType.CREATE);
+        managedBridgeService.createManagedBridge(bridgeDTO);
+
+        waitUntilManagedBridgeExists(bridgeDTO);
```

Comment:
I don't understand why we need to wait it here?
We are just creating namespace, secret and ManagedBridge CRD instance. There is no asyn operator here.

Reply:
Done.